### PR TITLE
Database/taskdaoimpl

### PR DESCRIPTION
**Pull request by Flozik**
In commit I describe all changes. Check the last one commit.
Two methods getAll and getTaskById don't work correctly and I don't know why.
After your review I will create comments and try to add logs.
image
method getTaskById return this. Why? I don't know

### CODE REVIEW

**Comment by komarichyn** [comment]
it is ok, you should translate your task object, just try to add method to string() for class Task

File: src/main/java/com/jc/tm/database/entity/Task.java [hunk 1]
```diff
@@ -6,19 +6,19 @@
 import java.util.List;
 
 public class Task {
-    private int id;
+    private long id;
```
Comment (komarichyn) [inline src/main/java/com/jc/tm/database/entity/Task.java:9]: for avoid addition autoboxing pls use Long instead long

File: src/main/java/com/jc/tm/database/dao/TaskDaoImpl.java [hunk 2]
```diff
@@ -1,5 +1,121 @@
...
+
+    //sql commands
+    private static final String INSERT_TASK = "INSERT INTO task (id, name, description, due_Dates, status) " +
+            "values (?, ?, ?, ?, ?)";
+    private static final String UPDATE_TASK = "UPDATE task SET name = ?, description = ?, due_Dates = ? where id = ?";
+    private static final String SELECT_ALL_TASK = "SELECT id, name, description, due_Dates, status FROM TASK";
+    private static final String SELECT_BY_ID_TASK = "select id, name, description, due_Dates, status from task where id = ?";
+    private static final String DELETE_TASK = "DELETE FROM task WHERE id = ?";
+
+    private static final String _id = "id";
```
Comment (komarichyn) [inline src/main/java/com/jc/tm/database/dao/TaskDaoImpl.java:24]: better way you use upper case for constants

File: src/main/java/com/jc/tm/database/dao/TaskDaoImpl.java [hunk 3]
```diff
@@ -1,5 +1,121 @@
...
+            //Display values
+            Task task = new Task();
+            task.setId(resultSet.getLong(_id));
+            task.setName(resultSet.getString(_name));
+            task.setDescription(resultSet.getString(_description));
+            task.setCreated(resultSet.getDate(_dueDates).toLocalDate());
+            task.setStatus(Status.valueOf(resultSet.getString(_status)));
+
+            taskList.add(task);
+            System.err.println("While loop end");
```
Comment (komarichyn) [inline src/main/java/com/jc/tm/database/dao/TaskDaoImpl.java:105]: do not use system.err here, better way to use log.error

**Review by komarichyn — CHANGES_REQUESTED** [review]
pls fix my notes

File: src/main/java/com/jc/tm/database/dao/TaskDaoImpl.java (hunk 2 above)
Comment (Flozik) [reply src/main/java/com/jc/tm/database/dao/TaskDaoImpl.java:24]: Sure

File: src/main/java/com/jc/tm/database/entity/Task.java (hunk 1 above)
Comment (Flozik) [reply src/main/java/com/jc/tm/database/entity/Task.java:9]: It is just mistake, because in other classes I use Long.

**Pull request merged**